templating/base_render_options.go

```diff
@@ -152,6 +159,54 @@ func LoadAndRenderSteps(template *Template, opts *BaseRenderOptions) (string, er
 	return rendered, nil
 }
 
+// ResolveAndMergeSecrets parses the secrets in the template, resolves them using vault providers and adds them to the values map.
+func resolveAndMergeSecrets(ctx context.Context, template *Template, secretResolver *SecretResolver, valueMap *Values) error {
```

Comment:
Do you need to pass *Values or maybe just Values? I also think return (Values, error) is more clear.